markdown/h2m/handlers/index.ts

```diff
@@ -82,7 +88,7 @@ export const handlers: QueryAndTransform[] = [
     },
     (node, t) =>
       h("heading", t(node, { shouldWrap: true, singleLine: true }), {
-        depth: Number(node.tagName.charAt(1)) || 1,
+        depth: (Number(node.tagName.charAt(1)) as 1 | 2 | 3 | 4 | 5) || 1,
```

Comment:
I think we need a `| 6` as well, for the `<h6>` tag

Reply:
The filter in line 85 only lists h1 to h5, that's why I omitted 6.